src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py

```diff
@@ -2809,9 +2809,13 @@ def test_vmss_run_command_e2e(self, resource_group, resource_group_location):
         })
         self.cmd('az network lb rule create -g {rg} --name allowTrafficRule --lb-name {lb_name} --backend-pool-name {lb_backend} --frontend-ip-name {lb_frontend} --backend-port 80 --frontend-port 80 --protocol tcp')
         public_ip = self.cmd('az network public-ip show --ids {lb_ip_id}').get_output_in_json()['ipAddress']
-        self.kwargs['instance_ids'] = " ".join(self.cmd('az vmss list-instances -n {vmss} -g {rg} --query "[].id"').get_output_in_json())
 
-        self.cmd('vmss run-command invoke --ids {instance_ids} --command-id RunShellScript --script "sudo apt-get update && sudo apt-get install -y nginx"')
+        instance_ids = self.cmd('az vmss list-instances -n {vmss} -g {rg} --query "[].id"').get_output_in_json()
+
+        for instance_ids in sorted(instance_ids):
```

Comment:
@adewaleo, do you know what was the root cause?

Reply:
No I do not for sure. As I couldn't reproduce the issue in python 2 and the [CI logs](https://travis-ci.org/Azure/azure-cli/jobs/515392771) don't give too much information. All I know is that:

- It occurs on python 2 test
- The command call I updated above fails with exit code 1

I am guessing that because we execute tasks [within the command concurrently](https://github.com/Azure/azure-cli/blob/421f863fe41229aa91bc3a5ee513a44b1558fe6b/src/azure-cli-core/azure/cli/core/commands/__init__.py#L317-L321) when multiple ids are passed, this can cause some sort of problems with rerunning the command from the recording.

The [vm version of this test ](https://github.com/Azure/azure-cli/blob/421f863fe41229aa91bc3a5ee513a44b1558fe6b/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py#L2774)hasn't had this flakiness issue and the only significant difference really is that the vmss version uses multiple ids (concurrent).